### PR TITLE
adding insertion sort for java

### PR DESCRIPTION
**Pull request by dpcoolmufa**
adding algorithms for java

### CODE REVIEW

File: Java/InsertionSort.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class Insertionsort{
```
Comment (AmitJoki) [inline Java/InsertionSort.java]: This should be InsertionSort

File: Java/InsertionSort.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    void insertsort(int arr[]){
```
Comment (AmitJoki) [inline Java/InsertionSort.java]: This should be insertSort.

File: Java/InsertionSort.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        for(int i=0;i<arr.length;i++){
+            int key=arr[i];
+            for(int j=i+1;j<arr.length;j++){
+                if(arr[j]<arr[i]){
+                    int temp=arr[j];
+                    arr[j]=arr[i];
+                    arr[i]=temp;
+                }
+            }
+
```
Comment (AmitJoki) [inline Java/InsertionSort.java]: These extra lines can be removed.

File: Java/InsertionSort.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+     public static void main(String[] args) {
+         int arr[]={5,4,3,2,1};
+        new Insertionsort().insertsort(arr);    
+    }
+
+    
+}
```
Comment (AmitJoki) [inline Java/InsertionSort.java:22]: Format the code so it is readable.

Comment (AmitJoki) [reply Java/InsertionSort.java:22]: Also, add '.class' files to .gitignore Class files aren't needed. For now, delete the .class file, add '.class' to .gitignore and push the code.

File: Java/Heapsort.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
+
+class HeapSort {
+    public int[] A;
+    public int heapSize; // mostly will give the index of last element in heap
+    // constructor
```
Comment (AmitJoki) [inline Java/Heapsort.java]: This is not needed.

File: Java/Heapsort.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
+
+class HeapSort {
+    public int[] A;
+    public int heapSize; // mostly will give the index of last element in heap
```
Comment (AmitJoki) [inline Java/Heapsort.java]: This one too, not very informative and misleading a bit. The comment can be removed.

File: Java/Heapsort.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        }
+        if ((largest<=heapSize) && (largest != i)) {
+            Swap(largest, i);
+            maxHeapify(largest);
+        }
+    }
+
+    void buildMaxHeap() {
+        for (int i = A.length / 2; i >= 0; i--) {
+            maxHeapify( i);
```
Comment (AmitJoki) [inline Java/Heapsort.java]: Use a formatter to format. There's a space before `i`.

File: Java/Heapsort.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
+
+class HeapSort {
+    public int[] A;
+    public int heapSize; // mostly will give the index of last element in heap
+    // constructor
+
+    void Heap(int[] array)
+    {
```
Comment (AmitJoki) [inline Java/Heapsort.java]: Use the same style throughout.
```
func () {

}
```

And use the this case - maxHeapify throughout for the methods. The Class name can be MergeSort but the methods should follow the naming convention of methods.

File: Java/Heapsort.java [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+        for (int i = heapSize; i > 0; i--) {
+            Swap(i,0);
+            heapSize--;
+            maxHeapify(0);
+        }
+        return A;
+    }
+
+    public static void main(String[] args) {
+        int arr[] = { 12, 63, 122, 674, 3, 7, 8433, 66, 32, 46, 7, 2, 234, 4};
```
Comment (AmitJoki) [inline Java/Heapsort.java]: Leave a space after 4. The simplest way to handle this is to use an IDE and its formatter option. There's no space between `heap` and `=`  in the following line.

**Review by AmitJoki — CHANGES_REQUESTED** [review]

File: Java/InsertionSort.java (hunk 1 above)
Comment (dpcoolmufa) [reply Java/InsertionSort.java]: got it!

**Comment by AmitJoki** [comment]
Also the comments should follow the grammar and start with the capital letter and all that stuff.

**Pull request merged**